hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/ColumnStatsIndexSupport.scala

```diff
@@ -455,10 +456,10 @@ object ColumnStatsIndexSupport {
       case w: LongWrapper => w.getValue
       case w: FloatWrapper => w.getValue
       case w: DoubleWrapper => w.getValue
+      case w: DecimalWrapper => w.getValue  // Moved above BytesWrapper to ensure proper matching
```

Comment:
I feel the strict ordering could be problematic. isn't there any other deterministic way to handle this?

Reply:
The issue is DecimalWrapper is a subtype of BytesWrapper - `{"name":"value","type":{"type":"bytes","logicalType":"decimal","precision":30,"scale":15}}`. So the BytesWrapper case is being triggered first. That is why this reordering. This reordering should not affect other types. Only those that have logical types as Decimal will get matched first.

Reply:
got it.